projects/igniteui-angular/src/lib/drop-down/drop-down.component.ts

```diff
@@ -345,24 +519,32 @@ export class IgxDropDownComponent extends IgxDropDownBase implements IDropDownBa
     public selectItem(newSelection?: IgxDropDownItemBase, event?: Event) {
         const oldSelection = this.selectedItem;
         if (!newSelection) {
-            newSelection = this._focusedItem;
+            newSelection = this.focusedItem;
         }
         if (newSelection === null) {
             return;
         }
-        if (newSelection.isHeader) {
+        if (newSelection instanceof IgxDropDownItemBase && newSelection.isHeader) {
             return;
         }
+        if (this.virtDir) {
+            newSelection = {
```

Comment:
This re-creation of the item shouldn't be required. Seems like it was due to the (now removed) union type.

Reply:
This is needed because **selectItem** is called with **this** inside of **IgxDropDownItem**'s click handler. If we do not explicitly recreate an object, when virtual, the drop down keeps a reference to the component instance instead - so when the data of the component instance changes, it stays selected, even if it is **not** bound to the correct data item